esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,233 @@
+        return cube.coord_dims(coord)
+
+    dims = {d for axis in axes for d in _get_dims_along_axis(cube, axis)}
+    return tuple(sorted(dims))
+
+
+class IrisESMFRegrid:
+    """:doc:`esmf_regrid:index` based regridding scheme.
+
+    Supports lazy regridding.
+
+    Parameters
+    ----------
+    method:
+        Either "conservative", "bilinear" or "nearest". Corresponds to the
+        :mod:`esmpy` methods
+        :attr:`~esmpy.api.constants.RegridMethod.CONSERVE`,
+        :attr:`~esmpy.api.constants.RegridMethod.BILINEAR` or
+        :attr:`~esmpy.api.constants.RegridMethod.NEAREST_STOD` used to
+        calculate regridding weights.
+    mdtol:
+        Tolerance of missing data. The value returned in each element of
+        the returned array will be masked if the fraction of masked data
+        exceeds ``mdtol``. ``mdtol=0`` means no missing data is tolerated while
+        ``mdtol=1`` will mean the resulting element will be masked if and only
+        if all the contributing elements of data are masked. If no value is
+        given, this will default to 1 for conservative regridding and 0
+        otherwise. Only available for methods 'bilinear' and 'conservative'.
+    use_src_mask:
+        If True, derive a mask from the source cube data,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368
+        <https://github.com/SciTools-incubator/iris-esmf-regrid/issues/368>`_
+        has been resolved.
+    use_tgt_mask:
+        If True, derive a mask from (first time step) of the target cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        ``'bilinear'`` and ``'conservative'`` and to :obj:`False` for method
+        ``'nearest'``. This default may be changed to :obj:`True` for all
+        schemes once `SciTools-incubator/iris-esmf-regrid#368`_ has been
+        resolved.
+    collapse_src_mask_along_axes:
+        When deriving the mask from the source cube data, collapse the mask
+        along the dimensions identified by these axes. Only points that are
+        masked at all time (``'T'``), vertical levels (``'Z'``), or both time
+        and vertical levels (``'TZ'``) will be considered masked.
+    collapse_tgt_mask_along_axes:
+        When deriving the mask from the target cube data, collapse the mask
+        along the dimensions identified by these axes. Only points that are
+        masked at all time (``'T'``), vertical levels (``'Z'``), or both time
+        and vertical levels (``'TZ'``) will be considered masked.
```

Comment:
If an axis exists, but is not mentioned here, the first slice along that axis will be used right? Would be good to mention that here.

Reply:
Yes. Do you think that is a safe option, also for ocean data? Or should we also allow e.g. coordinate names in addition to axes `'T'` and `'Z'`?

Reply:
It makes certainly sense in some use cases, as long as it's documented properly. And yes, allowing coordinate names in addition to axes sounds great!

Reply:
Done in cfc4330